src/enzo/Grid_DepositBaryons.C

```diff
@@ -213,8 +213,10 @@ int grid::DepositBaryons(grid *TargetGrid, FLOAT DepositTime)
        velocity field. */
  
 //  fprintf(stderr, "Grid_DepositBaryons - call dep_grid_cic\n");
-
-    float *input_density = BaryonField[DensNum];
+    int c_size = 1;
+    for (dim = 0; dim < GridRank; dim++) c_size *= GridDimension[dim];
+    float *input_density = new float[c_size];
+    //float *input_density = BaryonField[DensNum];
```

Comment:
I believe replacing `float * input_density = BaryonField[DensNum]` with the `new` assignment breaks this routine. It looks like `input_density` never gets assigned the values from `BaryonField[DensNum]`, except when its done in the `RK2SecondStepBaryonDeposit == 1` if statement below (but also see my comment below). I'm not sure if this is a problem, but I think it probably is. This means when `RK2SecondStepBaryonDeposit == 0`, `input_density` contains uninitialized values, making the `+=` later a problem.

The fix should just be adding in: `for (int i = 0; i < c_size; i++) input_density[i] = BaryonField[DensNum][i];`

Reply:
fixed